internal/sipsource/bucket_source.go

```diff
@@ -0,0 +1,101 @@
+package sipsource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.artefactual.dev/tools/bucket"
+	"gocloud.dev/blob"
+)
+
+const defaultLimit = 100
+
+// SIPBucketSource represents a SIP source that uses a cloud storage bucket
+// to store SIPs. It implements the SIPSource interface.
+type SIPBucketSource struct {
+	// ID is the unique identifier for the SIP source.
+	ID uuid.UUID
+	// Bucket is the bucket where SIPs are stored.
+	Bucket *blob.Bucket
+	// Name is the human-readable name of the SIP source.
+	Name string
+}
+
+var _ SIPSource = (*SIPBucketSource)(nil)
+
+// NewBucketSource creates a new SIPSource from the provided configuration.
+func NewBucketSource(ctx context.Context, cfg *Config) (*SIPBucketSource, error) {
+	if cfg.IsEmpty() {
+		// Return an empty SIP source if the configuration is empty.
+		return &SIPBucketSource{}, nil
+	}
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+
+	bucket, err := bucket.NewWithConfig(ctx, cfg.Bucket)
```

Comment:
We should support Azure buckets here, I'll follow up about that in another comment.